archai/common/azureml_helper.py

```diff
@@ -0,0 +1,132 @@
+from typing import Dict, Union
+from pathlib import Path
+
+from azure.ai.ml import MLClient, command, load_job
+from azure.ai.ml.entities import AmlCompute, Environment, Command, Job
+from azure.identity import DefaultAzureCredential
+
+def get_aml_client_from_file(config_path: Union[str, Path] = "../.azureml/config.json") -> MLClient:
+    credential = DefaultAzureCredential()
+    config_path = Path(config_path)
+
+    ml_client = MLClient.from_config(
+        credential=credential,
+        path=config_path
+    )
+    return ml_client
+
+
+def create_compute_cluster(
+        ml_client: MLClient,
+        cpu_compute_name: str,
+        type: str = "amlcompute",
+        size: str = "Standard_D14_v2",
+        min_instances: int = 0,
+        max_instances: int = 4,
+        idle_time_before_scale_down: int = 180,
+        tier: str = "Dedicated",
+        **kwargs):
+    try:
+        cpu_cluster = ml_client.compute.get(cpu_compute_name)
+        print(f"You already have a cluster named {cpu_compute_name}, we'll reuse it as is.")
+    except Exception:
+        cpu_compute = AmlCompute(
+            name=cpu_compute_name,
+            type=type,
+            # VM Family
+            size=size,
+            # Minimum running nodes when there is no job running
+            min_instances=min_instances,
+            # Nodes in cluster
+            max_instances=max_instances,
+            # How many seconds will the node running after the job termination
+            idle_time_before_scale_down=idle_time_before_scale_down,
+            # Dedicated or LowPriority. The latter is cheaper but there is a chance of job termination
+            tier=tier,
+            **kwargs
+        )
+
+        cpu_cluster = ml_client.compute.begin_create_or_update(cpu_compute).result()
+        print(f"AMLCompute with name {cpu_cluster.name} is created, the compute size is {cpu_cluster.size}")
+
+
+def create_environment_from_file(
+        ml_client: MLClient,
+        custom_env_name: str = "aml-archai",
+        description: str = "Custom environment for Archai",
+        tags: dict = {"archai": "1.0.0"},
```

Comment:
Personally, I don't like pre-defined lists or dicts. For example:

```
tags: Optional[Dict[str, Any]] = None,
...
tags = tags or {"archai": "1.0.0"}
...
```

But @piero2c and @lovettchris should give their inputs/preferences as well.

Reply:
Done.